clang/test/SemaSYCL/sycl-restrict.cpp

```diff
@@ -189,7 +189,7 @@ struct trickyStruct {
 
 // function return type and argument both unsupported
 // expected-note@+1 2{{'commitInfraction' defined here}}
-__int128 commitInfraction(__int128 a) {
+[[intel::device_indirectly_callable]] __int128 commitInfraction(__int128 a) {
```

Comment:
Could you please explain why the attribute needs to add in this test case?

Reply:
Because somewhere in the test address of this function is taken and the flag `-fsycl-allow-func-ptr` is enabled, this is the case we want to diagnose. This test intended to check other diagnostics, not the one I'm adding, so I've added an attribute to not emit an additional error here. Same works for other function in this test.

Reply:
Thanks for the explanation.